### PR TITLE
[EVM-Equivalent-Yul] Add mload, mstore and mstore8 opcodes

### PR DESCRIPTION
**Pull request by ManuelBilbao**
# What ❔

This PR adds `mload`, `mstore` and `mstore8` opcodes, including their gas charges.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

**Comment by ManuelBilbao** [comment]
This contains an implementation for memory expansion as well as #343. We should check which should we use

File: system-contracts/contracts/EvmInterpreter.yul [hunk 1]
```diff
@@ -182,6 +182,32 @@ object "EVMInterpreter" {
                 gasRemaining := sub(prevGas, toCharge)
             }
 
+            function roundUp(value, multiple) -> rounded {
+                let reminder := mod(value, multiple)
```
Comment (gianbelinche) [inline system-contracts/contracts/EvmInterpreter.yul:186]: Fix typo, remainder

File: system-contracts/contracts/EvmInterpreter.yul [hunk 2]
```diff
@@ -344,6 +370,45 @@ object "EVMInterpreter" {
...
+                    evmGasLeft := chargeGas(evmGasLeft, 2)
+                }
+                case 0x51 { // OP_MLOAD
+                    let offset
+
+                    offset, sp := popStackItem(sp)
+
+                    let expansionGas := expandMemory(add(offset, 32))
+
+                    sp := pushStackItem(sp, mload(add(MEM_OFFSET_INNER(), offset)))
```
Comment (gianbelinche) [inline system-contracts/contracts/EvmInterpreter.yul:387]: Maybe separate this for better readability

**Review by gianbelinche — CHANGES_REQUESTED** [review]
Remember to not use this memory expansion function, instead use the one in #343 

Everything ese is Ok

File: system-contracts/contracts/EvmInterpreter.yul (hunk 1 above)
Comment (ManuelBilbao) [reply system-contracts/contracts/EvmInterpreter.yul]: Was deleted

File: system-contracts/contracts/EvmInterpreter.yul (hunk 2 above)
Comment (ManuelBilbao) [reply system-contracts/contracts/EvmInterpreter.yul]: Done

**Review by fborello-lambda — APPROVED** [review]

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**